### PR TITLE
feat(ST-179): add test for auth/register/teacher endpoint

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
- Updated `auth.e2e-spec.ts`  file to include test for `auth/register/teacher` endpoint
- Updated `auth.helpers.ts` file to include `createUniqueCode` and `createTeacherRegistrationData`

## Associated Ticket Link(s)

- https://trello.com/c/fxQLvPBO

## Screenshots / Screen Recordings
N/A

### CODE REVIEW

File: test/auth/auth.e2e-spec.ts [hunk 1]
```diff
@@ -66,4 +72,88 @@ describe("AuthController (e2e)", () => {
...
+    it("returns BAD_REQUEST(400) when unique code is invalid", async () => {
+      const teacherData = createTeacherRegistrationData();
+
+      const response = await request(httpServer)
+        .post("/auth/register/teacher")
+        .send(teacherData)
+        .expect(HttpStatus.BAD_REQUEST);
+
+      expect(response.body).toHaveProperty("message");
+      expect(response.body.message).toContain("Invalid unique code");
```
Comment (ishmam-iqbal-sazim) [inline test/auth/auth.e2e-spec.ts]: Shouldn't you also check if the uniqueCode usageCount decremented by 1?

File: test/auth/auth.e2e-spec.ts [hunk 2]
```diff
@@ -66,4 +72,88 @@ describe("AuthController (e2e)", () => {
...
+    });
+
+    it("returns BAD_REQUEST(400) when unique code has been used 3 times", async () => {
+      const teacherData = createTeacherRegistrationData();
```
Comment (ishmam-iqbal-sazim) [inline test/auth/auth.e2e-spec.ts]: Can put this in a beforeEach inside the describe block

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

File: test/auth/auth.e2e-spec.ts (hunk 1 above)
Comment (ghost) [reply test/auth/auth.e2e-spec.ts]: So I have added a test when the unique code is valid but the email is not. That's when the response will return usage.

**Pull request merged**